packages/playground/remote/src/lib/worker-thread.ts

```diff
@@ -186,6 +186,86 @@ export class PlaygroundWorkerEndpoint extends PHPWorker {
 	async replayFSJournal(events: FilesystemOperation[]) {
 		return replayFSJournal(this.__internal_getPHP()!, events);
 	}
+
+	async downloadWordPressAssets() {
+		await downloadWordPressAssets(this.__internal_getPHP()!);
+	}
+}
+
+async function downloadWordPressAssets(php: PHP) {
+	if (!php.requestHandler) {
+		logger.warn('No PHP request handler available');
+		return;
+	}
+
+	try {
+		const remoteAssetListPath = joinPaths(
+			php.requestHandler.documentRoot,
+			'wordpress-remote-asset-paths'
+		);
+		// If the remote asset list is not available this means that the WordPress static assets are already downloaded.
+		if (!php.fileExists(remoteAssetListPath)) {
+			return;
+		}
+		const wpVersion = await getLoadedWordPressVersion(php.requestHandler);
+		const staticAssetsDirectory =
+			wpVersionToStaticAssetsDirectory(wpVersion);
+		if (!staticAssetsDirectory) {
+			return;
+		}
+		const response = await fetch(
+			[
+				wordPressSiteUrl,
+				staticAssetsDirectory,
+				'wordpress-static.zip',
+			].join('/')
+		);
+
+		if (!response.ok) {
+			logger.warn('Failed to download WordPress assets');
+			return;
+		}
+
+		const zipPath = '/tmp/wordpress-static-assets.zip';
+		await php.writeFile(
+			zipPath,
+			new Uint8Array(await response.arrayBuffer())
+		);
+		await php.run({
+			code: `<?php
+				$document_root = ${phpVar(php.requestHandler.documentRoot)};
+				$zip_path = ${phpVar(zipPath)};
+				$zip = new ZipArchive;
+				$res = $zip->open($zip_path);
+				if ($res !== TRUE) {
+					return;
+				}
+				for ($i = 0; $i < $zip->numFiles; $i++) {
+					$filename = $zip->getNameIndex($i);
+					$extractPath = str_replace('wordpress-static', $document_root, $filename);
+					// Create directories if they don't exist
+					if (substr($filename, -1) === '/') {
+						if (is_dir($extractPath)) {
+							continue;
+						}
+						mkdir($extractPath, 0777, true);
+					} else {
+						// Extract files
+						copy("zip://$zip_path#".$filename, $extractPath);
+					}
+				}
+				$zip->close();
+			`,
+		});
+		if (await php.fileExists(zipPath)) {
+			await php.unlink(zipPath);
+		}
```

Comment:
I think this can be replaced by the `unzipFile` helper:

```suggestion
unzipFile(
    php.requestHandler.documentRoot,
    new File([ new Uint8Array(await response.arrayBuffer()) ], "assets.zip")
);
```

Reply:
I tried, but it didn't work. I also couldn't get it to work with just unzip in PHP or JS.
The only way unzipping works for me is file by file, but I might be doing something wrong.

Reply:
This seems to be the problem:

```
$extractPath = str_replace('wordpress-static', $document_root, $filename);
```

If `wp-content`, `wp-include` etc was already at the zip root, you could unzip it in `/wordpress` and the helper would do the right thing.